### PR TITLE
PR: Add JavaScript validation for Delete Terms modules #530

### LINKED ISSUE #530: Add JavaScript validation for Delete Terms modules
- [x] For Delete Terms by Name module, the term name can't be null
- [x] For Delete Terms by Post count module, the post count can't be null and should also be an int

### PR DESCRIPTION
**Pull request by esudalaimadan**
Adds Javascript Validation for Delete Terms by name and post count modules

Fix #530

### CODE REVIEW

File: assets/js/src/bulk-delete.js [hunk 1]
```diff
@@ -216,6 +216,14 @@ jQuery(document).ready(function () {
 	BulkWP.validateUserRole = function(that) {
 		return (null !== jQuery(that).parent().prev().find(".enhanced-role-dropdown").val());
 	};
+
```
Comment (sudar) [inline assets/js/src/bulk-delete.js]: Let's move these two functions to the `assets/js/src/delete-terms.js` file.

That way we can keep all delete terms related functions in a single place.

File: include/Core/Terms/Modules/DeleteTermsByNameModule.php [hunk 2]
```diff
@@ -43,7 +43,9 @@ public function render() {
 	}
 
 	public function filter_js_array( $js_array ) {
-		$js_array['validators'][ $this->action ] = 'noValidation';
+		$js_array['validators'][ $this->action ] = 'validateTermName';
+		$js_array['error_msg'][ $this->action ]  = 'enterTermName';
+		$js_array['msg']['enterTermName']        = __( 'Please enter valid Term Name', 'bulk-delete' );
```
Comment (sudar) [inline include/Core/Terms/Modules/DeleteTermsByNameModule.php]: Let's change this message to `Please enter the term name that should be deleted`

File: include/Core/Terms/Modules/DeleteTermsByPostCountModule.php [hunk 3]
```diff
@@ -35,17 +35,17 @@ public function render() {
...
 		$this->render_submit_button();
 	}
 
 	public function filter_js_array( $js_array ) {
-		$js_array['validators'][ $this->action ] = 'validatePostTypeSelect2';
-		$js_array['error_msg'][ $this->action ]  = 'selectPostType';
-		$js_array['msg']['selectPostType']       = __( 'Please select at least one post type', 'bulk-delete' );
+		$js_array['validators'][ $this->action ] = 'validatePostCount';
+		$js_array['error_msg'][ $this->action ]  = 'validPostCount';
+		$js_array['msg']['validPostCount']       = __( 'Please enter valid post count(Greater than or Equal to zero)', 'bulk-delete' );
```
Comment (sudar) [inline include/Core/Terms/Modules/DeleteTermsByPostCountModule.php]: Let's change this message to `Please enter the post count based on which terms should be deleted. A valid post count will be greater than or equal to zero`

File: assets/js/src/bulk-delete.js (hunk 1 above)
Comment (esudalaimadan) [reply assets/js/src/bulk-delete.js]: Fixed it.

File: include/Core/Terms/Modules/DeleteTermsByNameModule.php (hunk 2 above)
Comment (esudalaimadan) [reply include/Core/Terms/Modules/DeleteTermsByNameModule.php]: Fixed.

File: include/Core/Terms/Modules/DeleteTermsByPostCountModule.php (hunk 3 above)
Comment (esudalaimadan) [reply include/Core/Terms/Modules/DeleteTermsByPostCountModule.php]: Fixed.

File: include/Core/Terms/Modules/DeleteTermsByPostCountModule.php [hunk 4]
```diff
@@ -35,17 +35,21 @@ public function render() {
 
 			<?php _e( 'Delete Terms if the post count is ', 'bulk-delete' ); ?>
 			<?php $this->render_number_comparison_operators(); ?>
-			<input type="number" name="smbd_<?php echo esc_attr( $this->field_slug ); ?>_post_count" placeholder="<?php _e( 'Post count', 'bulk-delete' ); ?>">
+			<input type="number" name="smbd_<?php echo esc_attr( $this->field_slug ); ?>" placeholder="<?php $markup = _e( 'Post count', 'bulk-delete' ); ?>" min="0">
```
Comment (sudar) [inline include/Core/Terms/Modules/DeleteTermsByPostCountModule.php]: Are you sure the assignment to ``$markup`` is correct as ``_e('Post count', 'bulk-delete')`` seems to always return ``null``.

**Comment by sudar** [comment]
@esudalaimadan 

I have left a single review comment.

Also the StyleCi checks are failing for this PR.

**Pull request merged**